### DISCUSSION TITLE
Customising the menu_items blade file?

### DISCUSSION THREAD

**Discussion by fronbow** [Laravel-Backpack/community-forum#994] [Q&A (Help)]
Hi All

Is there any way of customising the `menu_items.blade.php` file?
I've already utilised the `@can` permissions for displaying purposes but I want to serve a different menu for a specific page?

Currently my theme is using a modified tabler with the vertical layout (vertical sidebar menu) but I have a page I want to serve that needs the menu to be at the top of the page layout with a subset of the existing menu? (It's to be displayed on a touch screen display, so screen real estate is crucial).

It would be nice if there was a way of having the `menu_items.blade.php` be a stub for pulling in other menus depending on scenario but that could be a possible feature request?!

Any help would be much appreciated :)

**Comment by karandatwani92** [comment]
Hey @fronbow 

Can you explain with graphics/screenshots? This will help me to understand the exact need!

So far, what I understood is
- You want to Switch the layout on a specific page
-- You will need to create a middleware similar to [this](https://github.com/Laravel-Backpack/demo/blob/d7fbb3ee33ea9175fdc2924b2b9738d51d5868d6/app/Http/Middleware/Theme.php#L20) one. Don't use the session; just change the condition to fit your specific need.
- You want to alter menu items  on a specific page.
-- Create two different menu blade files, and `@include()` them in the `menu_items.blade.php` conditionally.

**Reply by fronbow** [reply]
Sorry for the late reply, been away from my desk, I ended up doing it the second way and did the conditional check for the page. Cheers @karandatwani92 :)
